### PR TITLE
[CP-2995] Added computed fields for entities

### PR DESCRIPTION
**Pull request by mkurczewski**
JIRA Reference: [CP-2995]

### :memo: Description ️

-

### :muscle: Checklist before requesting a review - nice to have

- getState function in async thunk actions is correctly typed
- redux selectors are used in components / prop drilling is reduce

### :exclamation: Checklist before merging a pull request

- change went through the QA process
- translations are updated in dedicated application
- [CHANGELOG.md](./CHANGELOG.md) is updated

[CP-2995]: https://appnroll.atlassian.net/browse/CP-2995?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by dkarski** [comment]
One other thought: I wonder if it might make sense to handle the entity mapping at the service layer to keep the data presentation logic contained there. This feels like more of an entity-specific concern rather than something tied to application behavior, and it might be better managed directly within the data model or configuration provided by the API.

File: libs/generic-view/store/src/lib/entities/helpers/compute-field.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+  const values = fields.reduce((acc: unknown[], field) => {
+    if (typeof field === "string") {
+      if (field.endsWith("[]")) {
+        return [...acc, ...(get(entity, field.slice(0, -2)) as unknown[])]
+      }
+      return [...acc, get(entity, field)]
+    }
+    return [...acc, computeField(entity, field)]
+  }, [])
```
[the remark this answers is not in the document]
Comment (mkurczewski) [reply libs/generic-view/store/src/lib/entities/helpers/compute-field.ts:36]: Right, it's needed only for the first accumulator modification. Changed in [e169e11](https://github.com/mudita/mudita-center/pull/2098/commits/e169e11cc4844729297270d99cccc24b7bc92df3)

File: libs/generic-view/store/src/lib/entities/helpers/compute-field.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+          return (
+            undefinedCondition &&
+            emptyStringCondition &&
+            emptyArrayCondition &&
+            emptyObjectCondition &&
+            zeroCondition &&
+            falseCondition
+          )
+        })
+        .map((value) => (value === undefined ? [] : value))
```
[the remark this answers is not in the document]
Comment (mkurczewski) [reply libs/generic-view/store/src/lib/entities/helpers/compute-field.ts]: It's looks like a leftover, removed in [e169e11](https://github.com/mudita/mudita-center/pull/2098/commits/e169e11cc4844729297270d99cccc24b7bc92df3)

**Comment by mkurczewski** [comment]
> One other thought: I wonder if it might make sense to handle the entity mapping at the service layer to keep the data presentation logic contained there. This feels like more of an entity-specific concern rather than something tied to application behavior, and it might be better managed directly within the data model or configuration provided by the API.

In this case some part of a logic from entity configuration would be handled on renderer part (validators, globalValidators) and some on main (computed fields). I'm afraid this would lead to harder maintenance as would require adding another layer of communication to achieve exactly the same results.

Unless there are no serious issues with current approach, I'd leave it as it is.

**Review by dkarski — APPROVED** [review]

**Pull request merged**